File: 62-unique-paths/unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        MEM = [[1 for y in range(m)] for x in range(m+n)]
        indexLast = 0
        for x in range(m+n - 1):
            for y in range(1,min(x,m)):
                MEM[(indexLast + 1)][y] = MEM[indexLast][y-1] + MEM[indexLast][y]
            #indexLast = (indexLast + 1) % 2
            indexLast = x
        print(MEM)
        return MEM[indexLast][m -1]

File: 62-unique-paths/unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        // This is n over/choose k with n == m + n and k = n
        vector<vector<int>> MEM(m,vector<int>(n, 1));
        for(int x = 1; x < m; ++x){
            for(int y = 1; y < n; ++y){
                MEM[x][y] = MEM[x-1][y] + MEM[x][y-1];
            }
        }
        return MEM[m - 1][n - 1];
    }
};